auth/tests/api/users.rs

```diff
@@ -344,6 +328,181 @@ mod needs_docker {
+                    .respond_with(ResponseTemplate::new(200).set_body_json(
+                        serde_json::from_str::<Value>(MOCKED_SUBSCRIPTIONS[0]).unwrap(),
+                    ))
+                    .mount(&app.mock_server)
+                    .await;
+
+                // POST /users/:account_name with valid JWT and the user upgraded to pro.
+                app.send_request(request).await
+            };
+
+        assert_eq!(response.status(), StatusCode::OK);
+
+        // --- Then we proceed with deleting the subscription item. ---
+
+        // In the process of adding the subscription item, the auth service will try to sync the tier,
+        // fetching the subscription from stripe.
+        //
+        // Our deletion handler logic will then fetch the subscription again, so we can see if it has
+        // an item with the metadata ID of the item we want to delete.
+        //
+        // We return a mocked subscription that has the RDS item we added earlier in the test, with
+        // the same ID. We will get the subscription item id from the subscription, and use it in
+        // the delete request.
+        Mock::given(method("GET"))
+            .and(bearer_token(STRIPE_TEST_KEY))
+            .and(path("/v1/subscriptions/sub_1Nw8xOD8t1tt0S3DtwAuOVp6"))
+            .respond_with(
+                ResponseTemplate::new(200)
+                    .set_body_json(serde_json::from_str::<Value>(MOCKED_SUBSCRIPTIONS[4]).unwrap()),
+            )
+            .mount(&app.mock_server)
+            .await;
+
+        // Extract the subscription item id from the mocked subscription.
+        let subscription_item_id = serde_json::from_str::<Value>(MOCKED_SUBSCRIPTIONS[4]).unwrap();
+        let subscription_item_id = subscription_item_id
+            .get("items")
+            .unwrap()
+            .get("data")
+            .unwrap()
+            .get(1)
+            .unwrap()
+            .get("id")
+            .unwrap()
+            .as_str()
+            .unwrap();
+
+        // We'll first try with a non-existent subscription item metadata id, which should still
+        // return a 200, since some resources will have been created before billing was implemented.
+        let request = Request::builder()
+            .uri("/users/subscription/items/non-existent-id")
+            .method("DELETE")
+            .header(AUTHORIZATION, format!("Bearer {}", token))
+            .body(Body::empty())
+            .unwrap();
+
+        let response = app.send_request(request).await;
+
+        assert_eq!(response.status(), StatusCode::OK);
+
+        let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        let response: SubscriptionResponse = serde_json::from_slice(&body).unwrap();
+
+        assert!(!response.success);
+        assert_eq!(
+            response.message,
+            format!("found no subscription item with the given metadata id: non-existent-id")
```

Comment:
**nit**: Personally in those cases I prefer to construct the expected error and get back the error message from it instead of duplicating. This avoids having to fix the test if we change the wording. What matters is that we've return the right error, not the message itself.

Reply:
Yup, that is much better, thanks!